### PR TITLE
feat: rich visual reports dashboard with four-tab UI and OLS forecast

### PR DESCRIPTION
**Pull request by studert**
## Summary

- Adds a four-tab reporting dashboard (Overview, Trends, Usage, Forecast) to `/reports`, replacing the flat page layout
- Implements OLS linear regression for budget forecasting in a pure server-side utility (`src/lib/forecast.ts`)
- Three new Server Action query functions for time-series spend, license utilization, and budget forecast data
- All Recharts chart components lazy-loaded via a single `next/dynamic({ ssr: false })` boundary to keep the main bundle under budget

## Changes

**Backend**
- `src/types/index.ts` — 9 new interfaces: `PeriodSpendPoint`, `ToolUtilization`, `BudgetForecast`, `ReportOverviewData`, `ForecastChartPoint`, `ToolSummaryItem`, `CircleReportItem`, `MonthlySpend`, `ForecastPoint`
- `src/lib/forecast.ts` — Pure OLS utility; handles `history.length < 3` guard, negative projection floor, integer rounding
- `src/actions/budget.ts` — `getBilledCostsTimeSeries()`, `getBudgetForecast()`
- `src/actions/assignments.ts` — `getLicenseUtilizationByTool()`

**Frontend**
- `src/components/reports/sparkline.tsx` — Inline 80×32 LineChart, `isAnimationActive={false}`
- `src/components/reports/trends-chart.tsx` — LineChart with 3m/6m/12m client-side range selector
- `src/components/reports/utilization-chart.tsx` — Vertical stacked BarChart (assigned vs. available seats)
- `src/components/reports/forecast-chart.tsx` — Composite LineChart (historical + projected) with budget ceiling `ReferenceLine`
- `src/components/reports/reports-charts-panel.tsx` — Single lazy-load boundary; renders full tab content per active tab
- `src/app/reports/reports-tab-bar.tsx` — Client tab bar; URL-reflected tab state via `router.replace(?tab=<value>)`
- `src/app/reports/page.tsx` — Reads `searchParams`, parallel fetches, assembles all data for `ReportsTabBar`
- `src/app/reports/loading.tsx` — Tab nav + chart-shaped skeleton placeholders

**Tests**
- `vitest.config.ts` — Unit test config
- `tests/unit/forecast.test.ts` — 12 unit tests covering OLS correctness, edge cases, status logic, integer guarantees

## Test plan

- [ ] Navigate to `/reports` — Overview tab active, summary cards + tables visible
- [ ] Click Trends tab — URL updates to `?tab=trends`, LineChart renders with range buttons
- [ ] Click Usage tab — UtilizationChart + usage table with "Show all" toggle
- [ ] Click Forecast tab — forecast cards + composite chart with projected line and budget ceiling
- [ ] Refresh at `?tab=usage` — correct tab remains active
- [ ] Navigate to `?tab=bogus` — falls back to Overview
- [ ] Run `pnpm test` — 12/12 unit tests pass
- [ ] Run `pnpm typecheck` — zero errors
- [ ] Run `pnpm lint` — zero warnings

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/lib/forecast.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+const MONTH_NAMES = ["Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec"];
+
+function parseMonthLabel(label: string): { year: number; month: number } | null {
+  const parts = label.split(" ");
+  if (parts.length !== 2) return null;
+  const monthIdx = MONTH_NAMES.indexOf(parts[0]);
+  const year = parseInt(parts[1], 10);
+  if (monthIdx === -1 || isNaN(year)) return null;
+  return { year, month: monthIdx };
+}
```
[the remark this answers is not in the document]
Comment (studert) [reply src/lib/forecast.ts:30]: Fixed in 0d25a4a. `parseMonthLabel` now supports both `MMM YYYY` and `Qn YYYY` formats. Quarterly labels map Q1→month 0, Q2→month 3, Q3→month 6, Q4→month 9 via a regex match, so forecast projections anchor correctly for both period types.

File: src/components/reports/reports-charts-panel.tsx [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+                <TableHeader>
+                  <TableRow>
+                    <TableHead>Circle</TableHead>
+                    <TableHead>Users</TableHead>
+                    <TableHead>Licenses</TableHead>
+                    <TableHead>Expected Cost</TableHead>
+                  </TableRow>
+                </TableHeader>
+                <TableBody>
+                  {circleReport
```
[the remark this answers is not in the document]
Comment (studert) [reply src/components/reports/reports-charts-panel.tsx]: Fixed in 0d25a4a. Changed to `[...circleReport].sort(...)` to avoid mutating the prop.

File: src/app/reports/page.tsx [hunk 3]
```diff
@@ -82,6 +95,41 @@ export default async function ReportsPage() {
...
+  let spendTrendPct = 0;
+  if (lastTwo.length === 2 && lastTwo[0].billedCents > 0) {
+    const diff = lastTwo[1].billedCents - lastTwo[0].billedCents;
+    spendTrendPct = (diff / lastTwo[0].billedCents) * 100;
+    spendTrend =
+      Math.abs(spendTrendPct) < 1
+        ? "flat"
+        : diff > 0
+          ? "up"
+          : "down";
```
[the remark this answers is not in the document]
Comment (studert) [reply src/app/reports/page.tsx]: Fixed in 0d25a4a. The trend now derives from `historicalPeriods` (filtered to periods with `billedCents > 0`) before taking `.slice(-2)`, so future zero-spend periods no longer skew the calculation.

File: src/components/reports/forecast-chart.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+          y={budgetCeilingCents}
+          stroke="var(--chart-5)"
+          strokeDasharray="4 2"
+          ifOverflow="extendDomain"
+          label={{
+            value: "Budget Ceiling",
+            position: "insideTopRight",
+            fontSize: 12,
+          }}
+        />
```
[the remark this answers is not in the document]
Comment (studert) [reply src/components/reports/forecast-chart.tsx:54]: Fixed in 0d25a4a. Renamed the prop to \ and the caller now passes \. The reference line label is updated to "Avg. Monthly Budget" so it's on the same scale as the per-period spend values.

File: src/app/reports/reports-tab-bar.tsx [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+          activeTab={activeTab}
+          overviewData={overviewData}
+          trendsData={trendsData}
+          usageData={usageData}
+          forecastData={forecastData}
+          toolSummary={toolSummary}
+          circleReport={circleReport}
+        />
+      </div>
+    </Tabs>
```
[the remark this answers is not in the document]
Comment (studert) [reply src/app/reports/reports-tab-bar.tsx:88]: Fixed in 0d25a4a. Added `TabsContent` import and wrapped `ReportsChartsPanel` in `<TabsContent value={activeTab}>`. Radix will now correctly wire `aria-controls` / `role="tabpanel"` between each trigger and the panel.

File: src/actions/budget.ts [hunk 6]
```diff
@@ -539,6 +540,64 @@ export async function getBudgetWithCosts(
...
+  try {
+    const budgetWithCosts = await getBudgetWithCosts(budgetId);
+    if (!budgetWithCosts) return [];
+    return budgetWithCosts.periods.map((p) => ({
+      month: p.periodLabel,
+      billedCents: p.billedTotalCents,
+      expectedCents: p.expectedSpendCents,
+      plannedCents: p.plannedAmountCents,
+      periodIndex: p.periodIndex,
+    }));
```
[the remark this answers is not in the document]
Comment (studert) [reply src/actions/budget.ts]: Fixed in 0d25a4a. `getBilledCostsTimeSeries` now filters to periods where `new Date(p.startDate) <= today` before mapping, so only current and past periods are returned.

File: src/components/reports/reports-charts-panel.tsx [hunk 7]
```diff
@@ -0,0 +1,462 @@
...
+
+function buildForecastChartData(
+  trendsData: PeriodSpendPoint[],
+  forecastData: BudgetForecast
+): ForecastChartPoint[] {
+  const historical: ForecastChartPoint[] = trendsData.map((p, i) => ({
+    month: p.month,
+    historical: p.billedCents,
+    projected:
+      i === trendsData.length - 1 && forecastData.projections.length > 0
```
[the remark this answers is not in the document]
Comment (studert) [reply src/components/reports/reports-charts-panel.tsx]: Fixed in 0d25a4a. `buildForecastChartData` now scans backwards through `trendsData` to find the last index with `billedCents > 0` and uses that as the connection point, so the projected line anchors at the last real billed period rather than the last array entry.

File: src/components/reports/reports-charts-panel.tsx [hunk 8]
```diff
@@ -0,0 +1,462 @@
...
+                <TableHeader>
+                  <TableRow>
+                    <TableHead>Tool</TableHead>
+                    <TableHead>Vendor</TableHead>
+                    <TableHead>Active Users</TableHead>
+                    <TableHead>Expected Cost</TableHead>
+                  </TableRow>
+                </TableHeader>
+                <TableBody>
+                  {toolSummary
```
[the remark this answers is not in the document]
Comment (studert) [reply src/components/reports/reports-charts-panel.tsx]: Fixed in 0d25a4a. Changed to `[...toolSummary].sort(...)` to avoid mutating the prop.

File: src/components/reports/forecast-chart.tsx [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        <XAxis dataKey="month" tick={{ fontSize: 12 }} />
+        <YAxis
+          tickFormatter={(v: number) => `$${(v / 100).toFixed(0)}`}
+          tick={{ fontSize: 12 }}
+        />
+        <ChartTooltip
+          content={
+            <ChartTooltipContent
+              formatter={(value) =>
+                `$${((value as number) / 100).toFixed(2)}`
```
[the remark this answers is not in the document]
Comment (studert) [reply src/components/reports/forecast-chart.tsx]: Fixed in 0d25a4a. The formatter now guards against null with `value == null ? null : ...` so null series values (used to break the line between historical and projected) do not render as $0.00.

File: src/lib/forecast.ts [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+  if (denom === 0) return { slope: 0, intercept: n > 0 ? sumY / n : 0 };
+  const slope = (n * sumXY - sumX * sumY) / denom;
+  const intercept = (sumY - slope * sumX) / n;
+  return { slope, intercept };
+}
+
+export function forecastBudget(options: ForecastOptions): BudgetForecast {
+  const { history, actualSpendToDateCents, budgetCeilingCents, today } = options;
+  const monthsToProject = Math.min(Math.max(options.monthsToProject ?? 3, 3), 6);
+
```
[the remark this answers is not in the document]
Comment (studert) [reply src/lib/forecast.ts:57]: Fixed in 0d25a4a. Added `totalPeriodsRemaining?: number` to `ForecastOptions`. The caller (`getBudgetForecast`) now passes `remainingPeriods.length`. In `forecastBudget`, a separate loop over `annualProjectionCount` computes `projectedRemainingCents` covering all remaining periods — the displayed `projections` array is still capped at 3-6 points for the chart.
